clang/test/3C/type_params.c

```diff
@@ -118,15 +118,16 @@ void f1(int **a, float **b) {
 }
 
 void f2(int **a, int **c) {
-// CHECK: void f2(_Ptr<_Ptr<int>> a, int **c) {
+// CHECK: void f2(_Ptr<_Ptr<int>> a, int **c : itype(_Ptr<_Ptr<int>>)) {
   int **b = test1(a);
   // CHECK: _Ptr<_Ptr<int>> b =  test1<_Ptr<int>>(a);
   int *d = test1(c);
   // CHECK: int *d = test1(c);
 }
 
 void deep(int ****v, int ****w, int ****x, int ****y, int ****z) {
-  // CHECK: void deep(_Ptr<_Ptr<_Ptr<int *>>> v, _Ptr<_Ptr<_Ptr<int *>>> w, _Ptr<_Ptr<_Ptr<int *>>> x, _Ptr<_Ptr<_Ptr<int *>>> y, _Ptr<_Ptr<_Ptr<int *>>> z) {
+  // CHECK: void deep(int ****v : itype(_Ptr<_Ptr<_Ptr<_Ptr<int>>>>), int ****w : itype(_Ptr<_Ptr<_Ptr<_Ptr<int>>>>), int ****x : itype(_Ptr<_Ptr<_Ptr<_Ptr<int>>>>), int ****y : itype(_Ptr<_Ptr<_Ptr<_Ptr<int>>>>), int ****z : itype(_Ptr<_Ptr<_Ptr<_Ptr<int>>>>)) {
+
   int ****u = test_many(v, w, x, y, z);
   // CHECK: _Ptr<_Ptr<_Ptr<int *>>> u =  test_many<_Ptr<_Ptr<int *>>>(v, w, x, y, z);
   ***w = (int*) 1;
```

Comment:
The change here suggests that the function `deep` was (mostly) considered safe to begin with and is now considered entirely unsafe. But only w is touched, so shouldn't the other parameters be considered safe?

Reply:
Yes, i think this does uncover an issue in the new itype code, but I think i've found a fix.